azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java

```diff
@@ -0,0 +1,429 @@
+
+         throw new IllegalStateException(illegalStateExceptionMessage);
+      }
+
+      final VMDeployment deployment = deployments.iterator().next();
+
+
+      return new NodeAndInitialCredentials<VMDeployment>(deployment, name,
+              LoginCredentials.builder().user(loginUser).identity(loginUser).password(loginPassword).authenticateSudo(true).build());
+   }
+
+   @Override
+   public Iterable<VMHardware> listHardwareProfiles() {
+
+      final List<VMHardware> hwProfiles = Lists.newArrayList();
+      final List<String> locationIds = Lists.newArrayList();
+
+      Iterable<Location> locations = listLocations();
+      for (Location location : locations){
+         locationIds.add(location.name());
+
+         Iterable<VMSize> vmSizes = api.getVMSizeApi(location.name()).list();
+
+         for (VMSize vmSize : vmSizes){
+            VMHardware hwProfile = new VMHardware();
+            hwProfile.name = vmSize.name();
+            hwProfile.numberOfCores = vmSize.numberOfCores();
+            hwProfile.osDiskSizeInMB = vmSize.osDiskSizeInMB();
+            hwProfile.resourceDiskSizeInMB = vmSize.resourceDiskSizeInMB();
+            hwProfile.memoryInMB = vmSize.memoryInMB();
+            hwProfile.maxDataDiskCount = vmSize.maxDataDiskCount();
+            hwProfile.location = location.name();
+            hwProfiles.add(hwProfile);
+         }
+
+      }
+
+      checkAndSetHwAvailability(hwProfiles, Sets.newHashSet(locationIds));
+
+      return hwProfiles;
+   }
+   private void checkAndSetHwAvailability(List<VMHardware> hwProfiles, Collection<String> locations) {
+      Multimap<String, String> hwMap = ArrayListMultimap.create();
+      for (VMHardware hw : hwProfiles) {
+         hwMap.put(hw.name, hw.location);
+      }
+
+      for (VMHardware hw : hwProfiles) {
+         hw.globallyAvailable = hwMap.get(hw.name).containsAll(locations);
+      }
+   }
+
+   private void getImagesFromPublisher(String publisherName, List<VMImage> osImagesRef, String location) {
+
+      OSImageApi osImageApi = api.getOSImageApi(location);
+
+      Iterable<Offer> offerList = osImageApi.listOffers(publisherName);
+
+      for (Offer offer : offerList) {
+         Iterable<SKU> skuList = osImageApi.listSKUs(publisherName, offer.name());
+
+         for (SKU sku : skuList) {
+            VMImage vmImage = new VMImage();
+            vmImage.publisher = publisherName;
+            vmImage.offer = offer.name();
+            vmImage.sku = sku.name();
+            vmImage.location = location;
+            osImagesRef.add(vmImage);
+         }
+      }
+   }
+
+   private List<VMImage> listImagesByLocation(String location) {
+      final List<VMImage> osImages = Lists.newArrayList();
+
+      getImagesFromPublisher("Microsoft.WindowsAzure.Compute", osImages, location);
+      getImagesFromPublisher("MicrosoftWindowsServer", osImages, location);
+      getImagesFromPublisher("Canonical", osImages, location);
```

Comment:
Avoid hardcoding this, and configure a comma-separated property instead, so users can customize the publishers? These three values should be preconfigured as a default in the azure api metadata.